### PR TITLE
#6 #7 points

### PR DESCRIPTION
**Pull request by lumc03**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,11 @@
   "homepage": "https://github.com/lumc03/workshop-api-testing-js#readme",
   "devDependencies": {
     "chai": "^4.3.6",
+    "http-status-codes": "^2.2.0",
     "mocha": "^10.0.0"
+  },
+  "dependencies": {
+    "axios": "^0.27.2",
+    "mochawesome": "^7.1.3"
```
Comment (holgiosalos) [inline package.json]: mochawesome should be a dev dependency, to fix that, run: `npm i -D mochawesome`

File: test/MyFirstApiConsume.test.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    //Head 
+    it('Consume Head Service', async () => {
+
+        const data = {
+            id: '3',
+        };
+        const response = await axios.head('https://httpbin.org', data);
+
+        expect(response.status).to.equal(StatusCodes.OK);
```
Comment (holgiosalos) [inline test/MyFirstApiConsume.test.js]: add an expect to verify response content is empty

File: test/MyFirstApiConsume.test.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            name: 'luisa',
+            age: 22,
+
+        };
+
+        const response = await axios.post('https://httpbin.org/post', config);
+
+        expect(response.status).to.equal(StatusCodes.OK);
+        expect(response.data).to.have.property('origin');
+
```
Comment (holgiosalos) [inline test/MyFirstApiConsume.test.js]: add another expect to verify that response contains the config data

File: test/MyFirstApiConsume.test.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        };
+
+        config = {
+            name: 'sofia'
+        }
+
+        const response = await axios.patch('https://httpbin.org/patch', data, config);
+
+        expect(response.status).to.equal(StatusCodes.OK);
+        expect(response.config.name).to.equal("sofia");
```
Comment (holgiosalos) [inline test/MyFirstApiConsume.test.js]: add another expect to verify that response contains the id passed on the data

File: test/MyFirstApiConsume.test.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        config = {
+            name: 'juana',
+            age: 21,
+        }
+
+        const response = await axios.put('https://httpbin.org/put', data, config);
+
+        expect(response.status).to.equal(StatusCodes.OK);
+        expect(response.config.name).to.equal("juana");
+        expect(response.config.age).to.equal(21);
```
Comment (holgiosalos) [inline test/MyFirstApiConsume.test.js]: add another expect to verify that response contains the id passed on the data

File: test/MyFirstApiConsume.test.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    //Get with Parameters
+    it('Consume GET Service with Query Parameters', async () => {
+        const data = {
+            id: '3',
+        };
+
+        const response = await axios.get('https://httpbin.org/get', data);
+
+        expect(response.status).to.equal(StatusCodes.OK);
+        expect(response.data).to.have.property('origin');
```
Comment (holgiosalos) [inline test/MyFirstApiConsume.test.js]: add another expect to verify that response contains the id passed on the data

File: test/MyFirstApiConsume.test.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        data = {
+            name: 'maria',
+            age: 20
+        }
+
+
+        const response = await axios.put('https://httpbin.org/put', data, { id: 3 });
+        expect(response.status).to.equal(StatusCodes.OK);
+        expect(response.config.id).to.equal(3);
+
```
Comment (holgiosalos) [inline test/MyFirstApiConsume.test.js]: add one expect to verify the data in response is the same that the `data` object you sent in request:
`expect(response.data.json).to.eql(data);`

File: test/MyFirstApiConsume.test.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        data = {
+            name: 'sofia',
+        };
+
+        const response = await axios.patch('https://httpbin.org/patch', data, config);
+
+        expect(response.status).to.equal(StatusCodes.OK);
+        expect(response.config.id).to.equal(3);
+
```
Comment (holgiosalos) [inline test/MyFirstApiConsume.test.js]: add one expect to verify the data in response is the same that the `data` object you sent in request:
`expect(response.data.json).to.eql(data);`

File: test/MyFirstApiConsume.test.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+            name: 'luisa',
+            age: 22
+
+        };
+
+        const response = await axios.post('https://httpbin.org/post', data, config);
+
+        expect(response.status).to.equal(StatusCodes.OK);
+        expect(response.config.id).to.equal(3);
+
```
Comment (holgiosalos) [inline test/MyFirstApiConsume.test.js]: add one expect to verify the data in response is the same that the `data` object you sent in request:
`expect(response.data.json).to.eql(data);`

**Comment by lumc03** [comment]
Hello everyone
I did the authentication point with github and it worked perfectly. But I'm having problems with the eslint dependency, it always throws me an error. I tried using several commands that came out in the documentation besides the -- -- fix, but it still doesn't work. I don't really know how to solve it, so I need some help.

image

**Review by holgiosalos — APPROVED** [review]

**Comment by holgiosalos** [comment]
> Hello everyone I did the authentication point with github and it worked perfectly. But I'm having problems with the eslint dependency, it always throws me an error. I tried using several commands that came out in the documentation besides the -- -- fix, but it still doesn't work. I don't really know how to solve it, so I need some help.
> 
> image

it seems that you already solved the issue, let me know if isn't solved yet

**Pull request merged**